src/tests/test_utils/common_test_utils/src/graph_comparator.cpp

```diff
@@ -274,15 +274,16 @@ class NodeAndOutputDescription {
         OPENVINO_THROW(ss.str());
     }
 
-    bool result_and_output_match(size_t num_iterations) const {
+    bool result_and_output_match(int64_t num_iterations) const {
         if (const auto concat_desciption = ov::as_type<const SubGraphOp::ConcatOutputDescription>(m_description)) {
             if (m_result->output(0).get_element_type() != m_output.get_element_type()) {
                 return false;
             }
 
             const auto& output_partial_shape = m_output.get_partial_shape();
             const auto& result_partial_shape = m_result->output(0).get_partial_shape();
-            if (result_partial_shape.is_dynamic() && output_partial_shape.is_dynamic()) {
+            if (output_partial_shape.is_dynamic() &&
+                (result_partial_shape.is_dynamic() || (result_partial_shape.is_static() && num_iterations == -1))) {
```

Comment:
Where this value is set to `-1` and why need to be changed to i64?
Maybe instead -1 it should be check if it is max of size_t?

Reply:
From [`SubGraphOp`](https://github.com/openvinotoolkit/openvino/blob/58b02fb88fa660468d281e751f117c51cb22932c/src/core/include/openvino/op/util/sub_graph_base.hpp#L131) definition - public API.